src/app/filelogger.cpp

```diff
@@ -31,32 +31,100 @@
 #include <chrono>
 
 #include <QDateTime>
+#include <QDebug>
 #include <QDir>
+#include <QHash>
 #include <QTextStream>
-#include <QVector>
 
 #include "base/global.h"
 #include "base/logger.h"
 #include "base/utils/fs.h"
+#include "base/utils/gzip.h"
 
 namespace
 {
     const std::chrono::seconds FLUSH_INTERVAL {2};
+
+    bool isObsolete(const QFileInfo &info, FileLogger::FileLogAgeType ageType, int age)
+    {
+        QDateTime modificationDate = info.lastModified();
+        switch (ageType)
+        {
+        case FileLogger::DAYS:
+            modificationDate = modificationDate.addDays(age);
+            break;
+        case FileLogger::MONTHS:
+            modificationDate = modificationDate.addMonths(age);
+            break;
+        default:
+            modificationDate = modificationDate.addYears(age);
+        }
+        return modificationDate <= QDateTime::currentDateTime();
+    }
+
+    bool compressBackupFile(const Path &sourcePath, const Path &destPath, int level, QString &msg)
+    {
+        QFile source {sourcePath.data()};
+        QFileInfo info {source};
+        const QDateTime atime = info.lastRead();
+        const QDateTime mtime = info.lastModified();
+        // It seems the created time can't be modified on UNIX.
+        const QDateTime ctime = info.birthTime();
+        const QDateTime mctime = info.metadataChangeTime();
+
+        if (!source.open(QIODevice::ReadOnly))
+        {
+            msg = FileLogger::tr("Can't open %1!").arg(sourcePath.data());
+            return false;
+        }
+
+        bool ok = false;
+        const QByteArray data = Utils::Gzip::compress(source.readAll(), level, &ok);
```

Comment:
Also just pointing out (and wasn't looking closely at code). Compression operation is slow so it is very preferable use its own thread.